### ISSUE TITLE
Web - Scan - App crashed after tap on notification received from User B

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#28932]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Log in in NewDot as User A (incognito window) and log in in regular browser as User B
2. User A: click on FAB icon and select request money option
3. Go to scan tab and add receipt
4. Select user B and click on request button
5. go to transaction thread and click on amount and enter amount
6. Click on notification

## Expected Result:
Chat conversation should be opened after click on notification

## Actual Result:
App crashed after tap on notification received from User B

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] Windows / Chrome
- [ ] MacOS / Desktop

**Version Number:** 1.3.78.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/b7c14335-06c1-4d1d-9d91-5284a732b080

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by bfitzexpensify** [comment]
Reproduced, sending this external.

**Comment by bfitzexpensify** [comment]
Couple of proposals ready for review @ArekChr

**Comment by ArekChr** [comment]
@tienifr As I understand, after fix, we will not send scan report notifications at all?

**Comment by madmax330** [comment]
Not overdue

**Comment by madmax330** [comment]
@Beamanator @cristipaval assigning you since you're on the dev team for wave 4 and this is a related bug per: https://expensify.slack.com/archives/C057F53HR40/p1698076065260889

**Comment by trjExpensify** [comment]
I was investigating [DB's report](https://expensify.slack.com/archives/C049HHMV9SM/p1698699280477699) of an app crash from notification and thought it could be this. I couldn't explicitly see the console log error mentioned here though, but reproducing it I see it's the same one he mentioned. Adding it here for reference and the breadcrumbs: 

<img width="1511" alt="image" src="https://github.com/Expensify/App/assets/16232057/418340f7-633f-4fb4-ba8f-b4a0d2c33557">

`Cannot read properties of undefined (reading 'text')`


P.S who's picking this up for the BE changes? 🙏

**Comment by cristipaval** [comment]
I'll start on this today

**Comment by cristipaval** [comment]
Update: I'm working on the Auth PR which fixes the data returned by the `OpenReport` command. It's a small fix but the logic in the OpenReport command is quite complex, so I am spending some time to better understand that logic to avoid unexpected side effects. The PR will be ready for review today, for sure.

**Comment by cristipaval** [comment]
[Here](https://github.com/Expensify/Auth/pull/9115)'s the Auth PR that fixes the [OpenReport](url) command.

I'll start now on fixing the Pusher updates to not send to the other users while the receipt is still being scanned.

**Comment by cristipaval** [comment]
[Here](https://github.com/Expensify/Web-Expensify/pull/39523) is the Web-E PR to avoid sending pusher updates to other users when the receipt is being scanned.

**Comment by cristipaval** [comment]
This is still being reviewed.

**Comment by bfitzexpensify** [comment]
Still under review

**Comment by cristipaval** [comment]
yes, under review. @youssef-lr was ooo but he is back today and he can review the [Web-E PR ](https://github.com/Expensify/Web-Expensify/pull/39523)

**Comment by cristipaval** [comment]
[Web-E PR](https://github.com/Expensify/Web-Expensify/pull/39523) in staging atm.

**Comment by cristipaval** [comment]
Same status.

**Comment by cristipaval** [comment]
> [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/39523) in staging atm.

Web-E is in prod now, removing HOLD

**Comment by cristipaval** [comment]
@tienifr is working on the [App PR](https://github.com/Expensify/App/pull/31481)

**Comment by cristipaval** [comment]
Any updates @tienifr ?

**Comment by trjExpensify** [comment]
👋 coming from here ([internal ref](https://expensify.slack.com/archives/C05RECHFBEW/p1700510149038429?thread_ts=1700510024.787639&cid=C05RECHFBEW)), we're not sure that Web PR fully fixed the problem or if there's something else in play. 

The whispered report preview component and a notification for "Split $0.00" is being seen/sent to the other participants of a chat when the receipt is still scanning: 

![Screenshot_20231120-125812](https://github.com/Expensify/App/assets/16232057/0eddb5ad-7018-4418-b8dd-02e89cdbf531)
![Screenshot_20231120-125755~2](https://github.com/Expensify/App/assets/16232057/6f0e3bb4-9673-4254-9518-2c7ed7a5f0ef)

Let's make sure we cover that case in the fix. Thanks!

**Comment by dylanexpensify** [comment]
bump @tienifr 🙇‍♂️

**Comment by cristipaval** [comment]
I'll have a look tomorrow as I am busy with critical stuff today.

**Comment by cristipaval** [comment]
Had issues with my local environment and took me some time to make push notifications work in my local environment. I'll continue on fixing this today.

**Comment by cristipaval** [comment]
[Web-E PR](https://github.com/Expensify/Web-Expensify/pull/39910) that fixes push notifications is now ready for review.

**Comment by cristipaval** [comment]
> [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/39910) that fixes push notifications is now ready for review.

This is currently in staging

**Comment by cristipaval** [comment]
> [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/39910) that fixes push notifications is now ready for review.

FYI: This is now in production.

We're only waiting for the [App PR](https://github.com/Expensify/App/pull/31481) to hit production.

**Comment by bfitzexpensify** [comment]
Payment summary:

Contributor: @tienifr to be paid $500 via Upwork ✅ 
C+: no payment required. 

@ArekChr please complete the BZ checklist when you get a moment - thanks!

**Comment by bfitzexpensify** [comment]
Bump on the BZ checklist @ArekChr - thank you!

**Comment by bfitzexpensify** [comment]
Another bump here thanks @ArekChr

**Comment by cristipaval** [comment]
It seems that @ArekChr has been ooo. He'll take care of the checklist on Monday.

**Comment by ArekChr** [comment]
- [x] Link to the PR: I couldn't find the PR that introduced this bug 
- [x] Link to comment: n/a
- [x] Link to discussion: n/a
- [x] Determine if we should create a regression test for this bug: I think we should add a regression test here.

**Regression Test Proposal**

1. Open an incognito window and log in as User A on the NewDot
2. In a regular browser window, log in as User B.
3. User A: click on the FAB icon and select the request money option
4. Go to the scan tab and add the receipt
5. Select user B and click on the request button
6. go to the transaction thread click on the amount and enter the amount
7. Click on notification
8. Ensure the chat opens correctly

Do we agree :+1: or :-1:

**Comment by bfitzexpensify** [comment]
Thanks @ArekChr. Regression steps proposed in https://github.com/Expensify/Expensify/issues/359435. 

We're all done here, closing this one out.
